cdap-ui/app/directives/flow-graph/flow.js

```diff
@@ -184,7 +199,14 @@ module.directive('myFlowGraph', function (d3, dagreD3, $state, $filter) {
          * Handles node click and sends to flowlet page.
          */
         function handleNodeClick(nodeId) {
-          $state.go('flows.detail.runs.detail.flowlets.detail', {flowletId: nodeId});
+          handleHideTip(nodeId);
+          var instance = instanceMap[nodeId];
+          if (instance.type === 'STREAM') {
+            $state.go('flows.detail.runs.tabs.status.streamsDetail', {streamId: nodeId});
+          } else {
+            $state.go('flows.detail.runs.tabs.status.flowletsDetail', {flowletId: nodeId});
```

Comment:
This was causing issues with workflow diagrams. This directive has elements specific for flows. It has to handle this state change a little bit more dynamic so that we can use it in workflows too. (On node click on a workflow diagram should probably take it to a workflows child state)